src/jemalloc.c

```diff
@@ -1081,6 +1082,44 @@ obtain_malloc_conf(unsigned which_source, char buf[PATH_MAX + 1]) {
 	return ret;
 }
 
+static void
+validate_hpa_settings(void) {
+	size_t hpa_threshold = fxp_mul_frac(HUGEPAGE, opt_hpa_opts.dirty_mult) +
```

Comment:
nit: `if (!hpa_supported() || !opt_hpa) return;` at the very beginning

This means you'd also have to set hpa:true in the new test (should be fine?).

Reply:
We set `opt_hpa = true` directly before the test starts. See the comment in the test case (which was really written by @davidtgoldblatt, I just copied it from the `hpa_background_thread` test case) for an explanation of why.

Reply:
I'd say also add the `opt_hpa` check in the first branch, after the `!hpa_supported()` condition.

Reply:
My apologies, I missed that in your first comment.